libraries/freertos_plus/standard/https/src/iot_https_client.c

```diff
@@ -2349,6 +2534,148 @@ IotHttpsReturnCode_t IotHttpsClient_SendSync(IotHttpsConnectionHandle_t connHand
 
 /*-----------------------------------------------------------*/
 
+IotHttpsReturnCode_t IotHttpsClient_WriteRequestBody(IotHttpsRequestHandle_t reqHandle, char *pBuf, uint32_t len, int isComplete)
+{
+    HTTPS_FUNCTION_ENTRY( IOT_HTTPS_OK );
+
+    HTTPS_ON_NULL_ARG_GOTO_CLEANUP(reqHandle);
+    /* This function is not valid for a synchronous response. Applications need to configure the request body in 
+       IotHttpsRequestInfo_t.pSyncInfo_t.reqData before calling IotHttpsClient_SendSync(). */
+    HTTPS_ON_ARG_ERROR_GOTO_CLEANUP(reqHandle->isAsync);
+    HTTPS_ON_ARG_ERROR_MSG_GOTO_CLEANUP(isComplete == 1,
+        IOT_HTTPS_NOT_SUPPORTED,
+        "isComplete must be 1 in IotHttpsClient_WriteRequestBody() for the current version of the HTTPS Client library.");
```

Comment:
This is good. Please update the documentation in the header as well.

Reply:
Updated.